sgrep_lint/sgrep_main.py

```diff
@@ -115,11 +114,25 @@ def sgrep_error_json_to_message_then_exit(
         )
 
 
+def yield_targeting_options(args: argparse.Namespace) -> Iterator[str]:
+    """Yields include/exclude CLI options to call semgrep with.
+
+    This is based on the arguments given to semgrep-lint.
+    """
+    for pattern in args.include:
+        yield from ["-include", pattern]
+    for pattern in args.exclude:
+        yield from ["-exclude", pattern]
+    for pattern in args.exclude_dir:
+        yield from ["-exclude-dir", pattern]
```

Comment:
Couldn't make up my mind whether this solution is elegant, or gratuitously magical.

Reply:
I think this is fine but model @brendongo as the gatekeeper of magic 🧙‍♂️